### ISSUE TITLE
[Jenkins] trigger a performance test on pull requests or downstream_tests

### ISSUE THREAD

**Issue by seantalts** [stan-dev/cmdstan#692]
There is some code to execute relative tests (ie tests of one cmdstan commit against another) here: https://github.com/stan-dev/performance-tests-cmdstan/blob/master/Jenkinsfile#L51
We want to do this automatically for pull requests anywhere in CmdStan, Stan, or Math, so we should probably put a new stage here https://github.com/stan-dev/cmdstan/blob/develop/Jenkinsfile#L49 that triggers when branch is not master or develop. That will then call the `PerformanceTestsCmdstan/downstream_tests` the way the Stan build does here https://github.com/stan-dev/stan/blob/develop/Jenkinsfile#L141

**Comment by seantalts** [comment]
#691 solves this, though it might need one more thing (noted in the review on the PR).

To hijack this issue a little - can we parse the comparative results and display them prominently on the pull request somehow? You can see the results I'm talking about if you click on the CmdStan build, click on the Upstream CmdStan Performance Tests stage, go to that downstream_tests build, and click on the output of the "test cmdstan develop against cmdstan pointer in this branch" http://d1m1s1b1.stat.columbia.edu:8080/blue/organizations/jenkins/CmdStan%20Performance%20Tests/detail/downstream_tests/4/pipeline/16 and the output is near the end,
```
+ ./comparePerformance.py develop_performance.csv performance.csv
('stat_comp_benchmarks/benchmarks/gp_pois_regr/gp_pois_regr.stan', 0.99)
('stat_comp_benchmarks/benchmarks/low_dim_corr_gauss/low_dim_corr_gauss.stan', 0.96)
('stat_comp_benchmarks/benchmarks/eight_schools/eight_schools.stan', 0.99)
('stat_comp_benchmarks/benchmarks/irt_2pl/irt_2pl.stan', 1.01)
('stat_comp_benchmarks/benchmarks/gp_regr/gp_regr.stan', 1.0)
('performance.compilation', 1.42)
('stat_comp_benchmarks/benchmarks/low_dim_gauss_mix_collapse/low_dim_gauss_mix_collapse.stan', 1.33)
('stat_comp_benchmarks/benchmarks/pkpd/one_comp_mm_elim_abs.stan', 0.99)
('stat_comp_benchmarks/benchmarks/sir/sir.stan', 0.94)
('stat_comp_benchmarks/benchmarks/pkpd/sim_one_comp_mm_elim_abs.stan', 1.17)
('stat_comp_benchmarks/benchmarks/garch/garch.stan', 1.08)
('stat_comp_benchmarks/benchmarks/gp_regr/gen_gp_data.stan', 1.0)
('stat_comp_benchmarks/benchmarks/arK/arK.stan', 0.92)
('stat_comp_benchmarks/benchmarks/arma/arma.stan', 1.01)
('stat_comp_benchmarks/benchmarks/low_dim_gauss_mix/low_dim_gauss_mix.stan', 0.97)
1.04474708074
```

**Comment by serban-nicusor-toptal** [comment]
I may have an idea, working on it.

**Comment by seantalts** [comment]
Fixed by #691
